### PR TITLE
Extract GetStatusInformation as a pure helper

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?
`GetStatusInformation` is renamed `GetInformation` and is now a pure helper. This prepare the path to migration metadata as a component.

### Describe how to test/QA your changes

Sanity check for each team. There should be no change in behavior.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

File: pkg/ebpf/bytecode/runtime/telemetry.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (tm *CompilationTelemetry) SubmitTelemetry(filename string, statsdClient st
 		platform = strings.ToLower(target.Distro.Display)
 	} else {
 		log.Warnf("failed to retrieve host platform information from nikos: %s", err)
-		platform = host.GetStatusInformation().Platform
+		platform = host.GetPlatformName()
```
Comment (brycekahle) [inline pkg/ebpf/bytecode/runtime/telemetry.go:85]: Does this need to be `hostMetadataUtils` now?

File: comp/metadata/host/utils/host_info_windows.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			info.BootTime = uint64(bootTime.Unix())
+			pids, _ := Pids()
+			info.Procs = uint64(len(pids))
+			info.OS = runtime.GOOS
+
+			info.KernelArch = runtime.GOARCH
+
+			pi := platform.CollectInfo()
+			info.Platform, _ = pi.OS.Value()
+			info.PlatformFamily, _ = pi.OS.Value()
```
Comment (pgimalac) [inline comp/metadata/host/utils/host_info_windows.go:70]: There is a `ValueOrDefault` function, and I guess using the `Family` field is more appropriate for the 2nd value
```suggestion
			info.Platform = pi.OS.ValueOrDefault()
			info.PlatformFamily = pi.Family.ValueOrDefault()
```

File: comp/metadata/host/utils/host_info_windows.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+			upTime := time.Duration(getTickCount64()) * time.Millisecond
+			bootTime := time.Now().Add(-upTime)
+			info.Uptime = uint64(upTime.Seconds())
+			info.BootTime = uint64(bootTime.Unix())
+			pids, _ := Pids()
+			info.Procs = uint64(len(pids))
+			info.OS = runtime.GOOS
+
+			info.KernelArch = runtime.GOARCH
```
Comment (pgimalac) [inline comp/metadata/host/utils/host_info_windows.go:66]: **💭 thought:** ‏Gohai has a `KernelRelease` field for Windows which could be used for this struct's `KernelVersion` maybe ?

**Review by pgimalac — APPROVED** [review]
I didn't initially notice that you mostly moved existing code, feel free to ignore my comments about moved code if you don't want to make changes

File: comp/metadata/host/utils/host_info_windows.go (hunk 3 above)
Comment (hush-hush) [reply comp/metadata/host/utils/host_info_windows.go:66]: Let do that in another PR. I don't want to introduce any change in behavior in the migration.

File: comp/metadata/host/utils/host_info_notwin.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+package utils
+
+import (
+	"github.com/shirou/gopsutil/v3/host"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GetInformation just returns an InfoStat object, we need some additional information that's not
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply comp/metadata/host/utils/host_info_notwin.go]: Good catch, the original comment was also missing the end . I updated it to have the same comment between windows and linux.

File: comp/metadata/host/utils/host_info_windows.go (hunk 2 above)
Comment (hush-hush) [reply comp/metadata/host/utils/host_info_windows.go:69]: Same as for the other comment, I don't want to introduce any behavior change in this PR.

**Review by kkhor-datadog — COMMENTED** [review]
Code changes look good, but adding some specific QA instructions would be helpful. For the process-agent we just need to check that the `running on platform` log line is still correctly showing the platform (this log line is shown when the process-agent is restarted).

**Pull request merged**